server/src/main/java/org/spine3/server/entity/EventDispatchingRepository.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2016, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity;
+
+import com.google.common.base.Optional;
+import com.google.protobuf.Message;
+import org.spine3.Internal;
+import org.spine3.base.Event;
+import org.spine3.base.EventContext;
+import org.spine3.base.Events;
+import org.spine3.server.BoundedContext;
+
+import java.util.Set;
+
+/**
+ * Abstract base for repositories that deliver events to entities they manage.
+ *
+ * @param <I> the type of IDs of entities
+ * @param <E> the type of entities
+ * @param <S> the type of entity state messages
+ * @author Alexander Yevsyukov
+ */
+public abstract class EventDispatchingRepository<I, E extends Entity<I, S>, S extends Message>
```

Comment:
As discussed vocally, the naming makes sense in terms of the proposed type hierarchy.

So, let's leave it as `EventDispatchingRepository`.